src/controllers/users.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const usersServices = require('../services/users');
+
+const getAllUsers = (req, res) => {
+  const users = usersServices.getAllUsers();
+
+  res.send(users);
+};
+
+const getUserById = (req, res) => {
+  const { userId } = req.params;
+
+  if (isNaN(parseInt(userId))) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const foundUser = usersServices.getUserById(userId);
+
+  if (!foundUser) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(foundUser);
+};
+
+const createUser = (req, res) => {
+  const { name } = req.body;
+
+  if (typeof name !== 'string' || !name.length) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newUser = usersServices.createUser(name);
+
+  res.statusCode = 201;
+  res.send(newUser);
+};
+
+const removeUser = (req, res) => {
+  const { userId } = req.params;
+
+  if (isNaN(parseInt(userId))) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const foundUser = usersServices.getUserById(userId);
+
+  if (!foundUser) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  usersServices.removeUser(userId);
+
+  res.sendStatus(204);
+};
+
+const updateUser = (req, res) => {
+  const { userId } = req.params;
+  const { name } = req.body;
+
+  if (isNaN(parseInt(userId))) {
```

Comment:
```suggestion
  if (!isValidNumericField(userId)) {
```
Move this logic to utility